lib/foreman_scc_manager/engine.rb

```diff
@@ -29,37 +29,48 @@ class Engine < ::Rails::Engine
 
         # Add permissions
         security_block :foreman_scc_manager do
-          permission :view_scc, {
-            :'foreman_scc_manager/scc_accounts' => [:show, :index],
-            :'api/v2/scc_accounts' => [:show, :index]
-          }
-          permission :use_scc, {
-            :'foreman_scc_manager/scc_accounts' => [:bulk_subscribe],
-            :'api/v2/scc_accounts' => [:bulk_subscribe]
-          }
-          permission :new_scc, {
-            :'foreman_scc_manager/scc_accounts' => [:new, :create],
-            :'api/v2/scc_accounts' => [:create]
-          }
-          permission :edit_scc, {
-            :'foreman_scc_manager/scc_accounts' => [:edit, :update],
-            :'api/v2/scc_accounts' => [:update]
-          }
-          permission :delete_scc, {
-            :'foreman_scc_manager/scc_accounts' => [:destroy],
-            :'api/v2/scc_accounts' => [:destroy]
-          }
-          permission :sync_scc, {
-            :'foreman_scc_manager/scc_accounts' => [:sync],
-            :'api/v2/scc_accounts' => [:sync]
-          }
+          permission :view_scc_accounts,
+                     { :scc_accounts => [:show, :index],
+                       :'api/v2/scc_accounts' => [:show, :index] },
+                     :resource_type => 'SccAccount'
+
+          permission :use_scc_accounts,
+                     { :scc_accounts => [:bulk_subscribe],
+                       :'api/v2/scc_accounts' => [:bulk_subscribe] },
+                     :resource_type => 'SccAccount'
+
+          permission :new_scc_accounts,
+                     { :scc_accounts => [:new, :create],
+                       :'api/v2/scc_accounts' => [:create] },
+                     :resource_type => 'SccAccount'
+
+          permission :edit_scc_accounts,
+                     { :scc_accounts => [:edit, :update],
+                       :'api/v2/scc_accounts' => [:update] },
+                     :resource_type => 'SccAccount'
+
+          permission :delete_scc_accounts,
+                     { :scc_accounts => [:destroy],
+                       :'api/v2/scc_accounts' => [:destroy] },
+                     :resource_type => 'SccAccount'
+
+          permission :sync_scc_accounts,
+                     { :scc_accounts => [:sync],
+                       :'api/v2/scc_accounts' => [:sync] },
+                     :resource_type => 'SccAccount'
         end
 
         # Add a new role called 'SccManager' if it doesn't exist
         role 'SccManager',
-             %i[view_scc use_scc new_scc edit_scc delete_scc sync_scc],
+             %i[view_scc_accounts use_scc_accounts new_scc_accounts edit_scc_accounts delete_scc_accounts sync_scc_accounts],
              'Role granting permissons to manage SUSE Subscriptions'
 
+        role 'SccViewer',
+             %i[view_scc_accounts use_scc_accounts sync_scc_accounts],
+             'Role granting permissons to view and use SUSE Subscriptions'
```

Comment:
Should we add the necessary katello-permissions `create_products` and `view_products` as well?
They are needed to create the Product and Repository, which is triggered by selecting the SUSE-Products in Foreman SCC Manager GUI.

Reply:
Good idea but I wouldn't do this. One plugin shouldn't 'use' the permissions of another plugin. Please correct me if this common in foreman/rails.

Reply:
In general it is probably a bad idea to have a lot of direct code dependencies between plugins, but I am wondering if the concrete case is not the exception here. Given that the `foreman_scc_manager` does not make any kind of sense without Katello, maybe it is OK (and even called for) in this case?

Reply:
IMHO a role bundled with the plugin should work when being assigned to a user.
In my test, the user could Sync and select SCC-Products, but submitting the selection resulted in failing Tasks because the Katello-products could not be created.

It would have been OK, if the user would not be able to select or submit the SCC-products for synchronization (e.g. greyed out button).
But I prefer that a role has a complete working use-case.
And remember: Katello always needs to be installed for foreman_scc_manger to be installed, so I do not see a case where the permissions will not be there.

Reply:
added the two katello permissions